DemocratizingData/democratizingdata-api
# Task/dda 0006  models and schema

## Overview:

Models, schemas and endpoints for all db tables

## Related Github Issues:

- [DDA-0006](https://github.com/DemocratizingData/democratizingdata-api/issues/0006)

## Summary of Changes:

- Abstract SQLAlchemy Base models and mixins for common fields
- Basemodels and mixins for pydantic schemas
- model
- agency_run
- dataset_alias
- affiliation
- author
- asjc
- publisher
- journal
- publication_author
- publication_asjc
- publication_dataset_alias
- pda_model
- author_affiliation
- issn

## Testing Steps:

1. Run unit tests and integration tests
2. Check localhost:8000/docs and visually inspect the generated documentation
3. Check each new endpoint (curl http://localhost:8000/journals?limit=10 etc)

## UI Photos:

## Notes:

- Some tables are currently failing integration tests due to incompatible data in the dev database
